src/main/java/io/github/jhipster/online/service/SubGenEventService.java

```diff
@@ -121,68 +113,10 @@ public void deleteByOwner(GeneratorIdentity owner) {
     }
 
     public List<TemporalCountDTO> getFieldCount(Instant after, SubGenEventType field, TemporalValueType dbTemporalFunction) {
-        CriteriaBuilder builder = entityManager.getCriteriaBuilder();
-        CriteriaQuery<RawSQLField> query = builder.createQuery(RawSQLField.class);
-        Root<SubGenEvent> root = query.from(SubGenEvent.class);
-        ParameterExpression<Instant> dateParameter = builder.parameter(Instant.class, QueryUtil.DATE);
-        ParameterExpression<String> typeParameter = builder.parameter(String.class, QueryUtil.TYPE);
-
-        query
-            .select(
-                builder.construct(
-                    RawSQLField.class,
-                    root.get(dbTemporalFunction.getFieldName()),
-                    root.get(SubGenEvent_.type),
-                    builder.count(root)
-                )
-            )
-            .where(
-                builder.greaterThan(root.get(SubGenEvent_.date).as(Instant.class), dateParameter),
-                builder.equal(root.get(SubGenEvent_.type), typeParameter)
-            )
-            .groupBy(root.get(SubGenEvent_.type), root.get(dbTemporalFunction.getFieldName()));
-
-        return entityManager
-            .createQuery(query)
-            .setParameter(QueryUtil.DATE, after)
-            .setParameter(QueryUtil.TYPE, field.getDatabaseValue())
-            .getResultList()
-            .stream()
-            .map(
-                entry ->
-                    new TemporalCountDTO(
-                        TemporalValueType.absoluteMomentToInstant(entry.getMoment().longValue(), dbTemporalFunction),
-                        entry.getCount()
-                    )
-            )
-            .collect(Collectors.toList());
+        return subGenEventStatisticsService.getFieldCount(after, field, dbTemporalFunction);
```

Comment:
if a seperate service is created better to call it directly from the controller instead of indirection here